lib/tasks/pipeline.js

```diff
@@ -93,11 +92,29 @@ module.exports = Task.extend({
 
     var pluginNameRegex = /^(ember\-cli\-deploy\-)(.*)$/;
     var name            = addon.name.match(pluginNameRegex)[2];
-    if (configuredPluginList && configuredPluginList.indexOf(name) === -1) {
-      return;
-    }
 
-    var deployPlugin = addon.createDeployPlugin({name: name});
+    var configuredNameRegex = /(.*?):(.*)/;
+    var configuredPluginHash = Object.create(null);
+    (deployConfig.plugins || []).forEach(function (pluginName) {
+      var split = pluginName.split(':');
+      if(split[0] === name) {
+        if(split[1]) {
+          configuredPluginHash[pluginName.match(configuredNameRegex)[2]] = split[0];
+        } else {
+          configuredPluginHash[split[0]] = split[0];
+        }
+      }
+    });
+
+    Object.keys(configuredPluginHash).forEach(function(configuredName) {
+      if(configuredPluginHash[configuredName] === name) {
+        self._createDeployPluginsForConfiguredName(addon, name, configuredName);
+      }
+    });
+  },
```

Comment:
As discussed, we need to support the no-whitelist autodiscovery path. I'm fine with those being two separate branches of code. I think that would make it more obvious what's going on and also help us provide more useful error messages when there is a whitelist entry for a plugin that is not available.